### PR TITLE
Clean up dependency specifications

### PR DESCRIPTION
**Pull request by bradenmacdonald**
Upgrade PRs like https://github.com/openedx/frontend-app-learning/pull/934 are currently blocked because this plugin is overly specifying its dependencies. There were also a number of other problems which I've fixed - see inline comments.

```
npm warn Could not resolve dependency:
npm warn peer redux@"4.1.2" from @edx/frontend-lib-learning-assistant@2.2.4
npm warn node_modules/@edx/frontend-lib-learning-assistant
npm warn   @edx/frontend-lib-learning-assistant@"^2.2.4" from the root project
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,53 +39,37 @@
...
+  "dependencies-note": "Most deps are peer deps because they're specified by frontend-app-learning.",
   "dependencies": {
     "@edx/brand": "npm:@edx/brand-openedx@1.2.0",
-    "@fortawesome/fontawesome-svg-core": "1.2.36",
-    "@fortawesome/free-brands-svg-icons": "5.15.4",
-    "@fortawesome/free-regular-svg-icons": "5.15.4",
-    "@fortawesome/free-solid-svg-icons": "5.15.4",
-    "@fortawesome/react-fontawesome": "0.2.0",
     "@optimizely/react-sdk": "^2.9.2",
-    "core-js": "^3.36.1",
```
Comment (bradenmacdonald) [inline package.json:50]: `core-js` is not needed in 2024.

File: package.json [hunk 2]
```diff
@@ -39,53 +39,37 @@
...
+    "react-router-dom": "^6",
+    "redux": "^4"
   },
   "devDependencies": {
-    "@edx/browserslist-config": "^1.1.1",
     "@edx/frontend-platform": "^8.0.0",
-    "@edx/reactifex": "^2.1.1",
-    "@openedx/frontend-build": "14.0.3",
-    "@openedx/paragon": "^22.2.0",
-    "@reduxjs/toolkit": "1.8.1",
```
Comment (bradenmacdonald) [inline package.json:73]: These deps are all specified by the learning MFE and are already in `peerDependencies` so they don't also need to be in `devDependencies`.

File: package.json [hunk 3]
```diff
@@ -39,53 +39,37 @@
...
     "husky": "7.0.4",
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
-    "react-redux": "7.2.9",
-    "react-router": "6.15.0",
-    "react-router-dom": "6.15.0",
-    "redux": "4.1.2",
-    "regenerator-runtime": "0.13.11",
```
Comment (bradenmacdonald) [inline package.json:88]: `regenerator-runtime` is not needed in 2024.

**Comment by bradenmacdonald** [comment]
@rijuma @alangsto Could you please take a look at this?

**Review by alangsto — APPROVED** [review]
This looks good to me, thank you!

**Comment by bradenmacdonald** [comment]
Thanks @alangsto! I will need someone on your side to merge this whenever ready.

**Comment by bradenmacdonald** [comment]
@alangsto Good point; changed to `fix` :)

**Pull request merged**